### ISSUE TITLE
'template' tag crashes when `pug` is used as the pre-html language

### ISSUE THREAD

**Issue by kzw** [vuejs/vue-test-utils#1417]
### Version
1.0.0-beta.31

### Reproduction link
[https://github.com/kzw/vuetest](https://github.com/kzw/vuetest)

### Steps to reproduce
I assume that `make` is available on your system.  At the root of the repo in master branch

npm i

./test

This test fails

In the branch `passing` the test passes.  You can examine the diff of `master` and `passing`.  In the `master` branch, if the `Home.vue `is rewritten with `html` instead of pug, the test passes.  

### What is expected?
test passing in master branch

### What is actually happening?
test failing

---
Please make it easy to report very simple issue; otherwise nobody will report anything and this project may die out

**Comment by lmiller1990** [comment]
Hi! I opened an issue regarding making reporting bugs more visible: #1418 . Thanks for going to the effort to report this. Most people just use vue-cli to make a minimal project.

Anyway, I tried running `npm i` and `./test`. I got an error:

```
(master) $ ./test
Mon 27 Jan 2020 11:45:28 AEST
make -C scss
sass site.scss ../css/site.css
Errno::ENOENT: No such file or directory @ rb_sysopen - ../css/site.css
  Use --trace for backtrace.
make[1]: *** [../css/site.css] Error 1
make: *** [code/src/css/custom.css] Error 2
```

What do I need to do?

Pug is definitely supported: See [here](https://github.com/vuejs/vue-jest#supported-template-languages). 

If you can help me run your project, I can try to figure out the problem - I have a feeling it's related to configuration.

**Comment by kzw** [comment]
Sorry you are having problem.  I cloned a new copy of my test repo to new folder and I did not find this problem.  The problem could be due to differences in `sass`.  

In any case, I pushed some fixes to both branches so that you should not have this problem any more.  Could you please give it a go again? 

> Hi! I opened an issue regarding making reporting bugs more visible: #1418 . Thanks for going to the effort to report this. Most people just use vue-cli to make a minimal project.
> 
> Anyway, I tried running `npm i` and `./test`. I got an error:
> 
> ```
> (master) $ ./test
> Mon 27 Jan 2020 11:45:28 AEST
> make -C scss
> sass site.scss ../css/site.css
> Errno::ENOENT: No such file or directory @ rb_sysopen - ../css/site.css
>   Use --trace for backtrace.
> make[1]: *** [../css/site.css] Error 1
> make: *** [code/src/css/custom.css] Error 2
> ```
> 
> What do I need to do?
> 
> Pug is definitely supported: See [here](https://github.com/vuejs/vue-jest#supported-template-languages).
> 
> If you can help me run your project, I can try to figure out the problem - I have a feeling it's related to configuration.

**Comment by kzw** [comment]
What the issue creating app does not ask is the stack traces I am getting.  Here it is 

` FAIL  __test__/main.test.js
  ● Test suite failed to run

    Jest encountered an unexpected token

    This usually means that you are trying to import a file which Jest cannot parse, e.g. it's not plain JavaScript.

    By default, if Jest sees a Babel config, it will use that to transform your files, ignoring "node_modules".

    Here's what you can do:
     • To have some of your "node_modules" files transformed, you can specify a custom "transformIgnorePatterns" in your config.
     • If you need a custom transformation specify a "transform" option in your config.
     • If you simply want to mock your non-JS modules (e.g. binary assets) you can stub them out with the "moduleNameMapper" config option.

    You'll find more details and examples of these config options in the docs:
    https://jestjs.io/docs/en/configuration.html

    Details:

    SyntaxError: Unexpected token (1:367)

      at Parser.pp$4.raise (node_modules/vue-template-es2015-compiler/buble.js:2757:13)
      at Parser.pp.unexpected (node_modules/vue-template-es2015-compiler/buble.js:647:8)
      at Parser.pp.expect (node_modules/vue-template-es2015-compiler/buble.js:641:26)
      at Parser.pp$2.parseBindingList (node_modules/vue-template-es2015-compiler/buble.js:1694:19)
      at Parser.pp$1.parseFunctionParams (node_modules/vue-template-es2015-compiler/buble.js:1231:22)
      at Parser.pp$1.parseFunction (node_modules/vue-template-es2015-compiler/buble.js:1218:8)
      at Parser.pp$3.parseExprAtom (node_modules/vue-template-es2015-compiler/buble.js:2184:17)
      at Parser.<anonymous> (node_modules/vue-template-es2015-compiler/buble.js:6003:24)
      at Parser.parseExprAtom (node_modules/vue-template-es2015-compiler/buble.js:6129:31)
      at Parser.pp$3.parseExprSubscripts (node_modules/vue-template-es2015-compiler/buble.js:2047:19)`



Also I have been using `pug` with Vue from the beginning (which is not very long) but only run into this error when I started to use this html tag `template` as recommended in `bootstrap-cue`.  The code appears to run fine, but this particular test (which I use to test the entry point code) starts to fail.  My current workarounds are 1) abandon pug for this component or 2) maintain pug file and javascript files separately and put them back together to create `.vue` file.

**Comment by lmiller1990** [comment]
I got this to work using [this](https://github.com/vuejs/vue-jest/issues/175#issuecomment-575043315) comment. Update your package.json. In your Jest key:

```json
"jest": {
"globals": {
      "vue-jest": {
        "pug": { "doctype": "html" }
      }
    }
}
```

Closing, post if this does not work for you. There are other warnings but unrelated to this.

BTW this is a super weird template - if you use vue cli to generate your project, everything will work out of the box.
